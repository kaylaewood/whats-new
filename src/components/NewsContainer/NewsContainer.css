.newsContainer {
  grid-area: news;
  column-count: 2;
  column-gap: 2em;
  padding: 0 2em;
}
