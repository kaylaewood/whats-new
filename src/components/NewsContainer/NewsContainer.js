import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

function NewsContainer(props) {
  return (
    <div className='newsContainer'>
      {props.news.map(article => {
        return <NewsArticle
          key = {article.id}
          headline = {article.headline}
          description = {article.description}
          link = {article.url}
          image = {article.img}
        />
      })}
    </div>
  )
}


export default NewsContainer;
