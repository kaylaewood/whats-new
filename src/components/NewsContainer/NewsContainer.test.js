import React from 'react';
import ReactDOM from 'react-dom';
import NewsContainer from './NewsContainer';
import { shallow } from 'enzyme';

describe('NewsContainer', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow(<NewsContainer
      news = {[{
        id: 2,
        headline: 'Headline',
        description: 'This is the description',
        link: 'This is a link url.',
        image: 'This is the image url.'
      },
      {
        id: 1,
        headline: 'Headline2',
        description: 'This is the other description',
        link: 'This is anoth link url.',
        image: 'This is the other image url.'
      }
    ]}
    />);
    expect(wrapper).toMatchSnapshot();
  })
});
