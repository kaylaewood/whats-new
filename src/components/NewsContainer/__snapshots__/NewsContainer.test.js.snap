// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should correctly match the snapshot with all the data passed correctly 1`] = `
<div
  className="newsContainer"
>
  <NewsArticle
    description="This is the description"
    headline="Headline"
    key="2"
  />
  <NewsArticle
    description="This is the other description"
    headline="Headline2"
    key="1"
  />
</div>
`;
